src/game/common/skirmishbattlehonors.cpp

```diff
@@ -27,3 +39,617 @@ int SkirmishBattleHonors::Get_Endurance_Medal(Utf8String name, int level) const
+    if (listbox == NULL) {
+        return;
+    }
+    SkirmishBattleHonors *BattleHonors = nullptr;
+    BattleHonors = new SkirmishBattleHonors();
+    listbox->Win_Set_Tooltip_Func(Battle_Honor_Tooltip);
+    Gadget_List_Box_Reset(listbox);
+
+    int honors = BattleHonors->Get_Honors();
+
+    key = g_theNameKeyGenerator->Name_To_Key("SkirmishGameOptionsMenu.wnd:StaticTextStreakValue");
+    GameWindow *window = g_theWindowManager->Win_Get_Window_From_Id(nullptr, key);
+    if (window != NULL) {
+        int WinStreak = BattleHonors->Get_Win_Streak();
+        Utf16String str;
+        str.Format(U_CHAR("%d"), WinStreak);
+        Gadget_Static_Text_Set_Text(window, str);
+    }
+
+    key = g_theNameKeyGenerator->Name_To_Key("SkirmishGameOptionsMenu.wnd:StaticTextBestStreakValue");
+    window = g_theWindowManager->Win_Get_Window_From_Id(nullptr, key);
+    if (window != 0) {
+        int bestWinStreak = BattleHonors->Get_Best_Win_Streak();
+        Utf16String str;
+        str.Format(U_CHAR("%d"), bestWinStreak);
+        Gadget_Static_Text_Set_Text(window, str);
+    }
+
+    key = g_theNameKeyGenerator->Name_To_Key("SkirmishGameOptionsMenu.wnd:StaticTextWinsValue");
+    window = g_theWindowManager->Win_Get_Window_From_Id(nullptr, key);
+    if (window != 0) {
+        int wins = BattleHonors->Get_Wins();
+        Utf16String str;
+        str.Format(U_CHAR("%d"), wins);
+        Gadget_Static_Text_Set_Text(window, str);
+    }
+
+    key = g_theNameKeyGenerator->Name_To_Key("SkirmishGameOptionsMenu.wnd:StaticTextLossesValue");
+    window = g_theWindowManager->Win_Get_Window_From_Id(nullptr, key);
+    if (window != 0) {
+        int losses = BattleHonors->Get_Losses();
+        Utf16String str;
+        str.Format(U_CHAR("%d"), losses);
+        Gadget_Static_Text_Set_Text(window, str);
+    }
+
+    Reset_Battle_Honor_Insertion();
+    Gadget_List_Box_Add_Entry_Image(listbox, 0, 0, 0, 10, 10, 1, -1);
+
+    int row = 1;
+    int column = 0;
+
+    const Image *image;
+    if (BattleHonors->Get_China_Campaign_Complete(DIFFICULTY_HARD)) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("ChinaCampaign_G");
+        Insert_Battle_Honor(listbox, image, 1, CAMPAIGN_CHINA, &row, &column, 0, 0);
+    } else if (BattleHonors->Get_China_Campaign_Complete(DIFFICULTY_NORMAL)) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("ChinaCampaign_S");
+        Insert_Battle_Honor(listbox, image, 1, CAMPAIGN_CHINA, &row, &column, 0, 0);
+    } else if (BattleHonors->Get_China_Campaign_Complete(DIFFICULTY_EASY)) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("ChinaCampaign_B");
+        Insert_Battle_Honor(listbox, image, 1, CAMPAIGN_CHINA, &row, &column, 0, 0);
+    } else {
+        image = g_theMappedImageCollection->Find_Image_By_Name("ChinaCampaign_B");
+        Insert_Battle_Honor(listbox, image, (char)(honors >> 0xC) & 1, CAMPAIGN_CHINA, &row, &column, 0, 0);
```

Comment:
I think this is meant to be written as

(honors & CAMPAIGN_CHINA) != 0

Several times in this file.

Reply:
Didn't realize they were the same with three different bases all over